apps/opik-frontend/src/api/projects/useProjectByName.ts

```diff
@@ -1,43 +1,29 @@
-import { QueryFunctionContext, useQuery } from "@tanstack/react-query";
+import { useQuery } from "@tanstack/react-query";
 import api, { PROJECTS_REST_ENDPOINT, QueryConfig } from "@/api/api";
 import { Project } from "@/types/projects";
 
 type UseProjectByNameParams = {
   projectName: string;
 };
 
-type UseProjectsListResponse = {
-  content: Project[];
-  total: number;
-};
-
-// @todo: replace it with another request to get a project id only when the be is ready
-const getProjectByName = async (
-  { signal }: QueryFunctionContext,
-  { projectName }: UseProjectByNameParams,
-) => {
-  const { data } = await api.get<UseProjectsListResponse>(
-    PROJECTS_REST_ENDPOINT,
+const getProjectByName = async ({ projectName }: UseProjectByNameParams) => {
+  const { data } = await api.post<Project>(
+    `${PROJECTS_REST_ENDPOINT}retrieve`,
     {
-      signal,
```

Comment:
@aadereiko I have a question: Why request without a signal? That means the request will not be canceled when the hook is destroyed. Am I missing something?

Reply:
yeah, automatically, i removed it becuase of the POST request. brought it back, thanks!